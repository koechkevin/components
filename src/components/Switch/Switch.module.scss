@import '../../styles/global';

.switch {
  &:global(.ant-switch) {
    height: 14px;
    width: 36px;
    background: $color-grey-lighter;
  }

  &:global(.ant-switch-checked) {
    background-color: $color-primary-light;
  }

  &:global(.ant-switch::after) {
    background: $color-white;
    width: 20px;
    height: 20px;
    margin-top: -6px;
    margin-left: -2px;
  }

  &:global(.ant-switch-checked::after) {
    background-color: $color-primary;
    margin-left: 0px;
    margin-right: -2px;
  }
}
