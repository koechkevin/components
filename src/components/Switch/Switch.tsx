import { Switch as AntdSwitch } from 'antd';
import { SwitchProps } from 'antd/lib/switch';
import React from 'react';

import styles from './Switch.module.scss';

const Switch: React.FC<SwitchProps> = (props) => {
  const { className, ...otherProps } = props;
  const classNames = [styles.switch, className ? className : ''].join(' ');

  return <AntdSwitch className={classNames} {...otherProps} />;
};

export default Switch;
