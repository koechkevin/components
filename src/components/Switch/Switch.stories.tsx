import { storiesOf } from '@storybook/react';
import React from 'react';

import { Center } from '../Center';
import Switch from './Switch';

storiesOf('Switch', module)
  .addDecorator((story) => <Center>{story()}</Center>)
  .add('Default', () => <Switch />);
