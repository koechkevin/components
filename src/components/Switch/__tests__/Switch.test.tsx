import { Switch as AntdSwitch } from 'antd';
import { SwitchProps } from 'antd/lib/switch';
import { mount, ReactWrapper } from 'enzyme';
import React from 'react';

import Switch from '../Switch';

describe('Tests for switch component', () => {
  const props: SwitchProps = { onChange: jest.fn() };
  const wrapper: ReactWrapper = mount(<Switch {...props} />);

  it('should tests for the mounting of the switch component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain antd switch component', () => {
    expect(wrapper.find(AntdSwitch).exists()).toEqual(true);
  });

  it('should should switch on the switch', () => {
    const switchBtn: ReactWrapper = wrapper.find('.ant-switch');
    switchBtn.simulate('click');

    expect(props.onChange).toHaveBeenCalledTimes(1);
  });

  it('it should switch off the switch', () => {
    const switchBtn: ReactWrapper = wrapper.find('.ant-switch-checked');
    switchBtn.simulate('click');

    expect(props.onChange).toHaveBeenCalledTimes(2);
  });
});
