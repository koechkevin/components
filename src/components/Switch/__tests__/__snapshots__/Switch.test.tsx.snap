// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for switch component should tests for the mounting of the switch component 1`] = `ReactWrapper {}`;
